apps/api-documenter/src/documenters/YamlDocumenter.ts

```diff
@@ -559,29 +560,27 @@ export class YamlDocumenter {
     apiItem: ApiClass | ApiInterface
   ): void {
     if (apiItem instanceof ApiClass) {
-      if (apiItem.extendsType) {
-        yamlItem.extends = [this._renderType(uid, apiItem.extendsType.excerpt)];
-        yamlItem.inheritance = this._renderInheritance(uid, [apiItem.extendsType]);
-      }
       if (apiItem.implementsTypes.length > 0) {
         yamlItem.implements = [];
         for (const implementsType of apiItem.implementsTypes) {
           yamlItem.implements.push(this._renderType(uid, implementsType.excerpt));
         }
       }
-    } else if (apiItem instanceof ApiInterface) {
+    }
+
+    if (ApiExtendsMixin.isBaseClassOf(apiItem)) {
       if (apiItem.extendsTypes.length > 0) {
         yamlItem.extends = [];
         for (const extendsType of apiItem.extendsTypes) {
           yamlItem.extends.push(this._renderType(uid, extendsType.excerpt));
         }
         yamlItem.inheritance = this._renderInheritance(uid, apiItem.extendsTypes);
       }
+    }
 
-      const typeParameters: IYamlParameter[] = this._populateYamlTypeParameters(uid, apiItem);
-      if (typeParameters.length) {
-        yamlItem.syntax = { typeParameters };
-      }
+    const typeParameters: IYamlParameter[] = this._populateYamlTypeParameters(uid, apiItem);
+    if (typeParameters.length) {
+      yamlItem.syntax = { typeParameters };
```

Comment:
This code was previously only called for `ApiInterface` items, I think this was a bug? Now it's called for both `ApiClass` and `ApiInterface` items.

Reply:
This was introduced by PR #1317.  The PR notes say that  "type parameters are supported by docfx" but doesn't give any reason why the YAML is only populated for interfaces.

@rbuckton was this simply an oversight?  If we start emitting type parameters for classes in DocFX YAML, is there any risk of a regression?

Reply:
I reverted this change from this PR because it's no longer really related (I'm not even touching `YamlDocumenter.ts` anymore. It'd still be nice to fix this, it looks like an oversight, I'll open a separate PR.